src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,254 @@
+  }, []);
+
+  // lock body on modal show
+  useEffect(() => {
+    instance?.on("show", () => disableBodyScroll(instance.$el, { reserveScrollBarGap: true }));
+    instance?.on("hide", () => enableBodyScroll(instance.$el));
+    return () => {
+      instance?.off("show");
+      instance?.off("hide");
+    };
+  }, [instance]);
+
+  useEffect(() => {
+    instance?.on("hide", () => onClose());
+    return () => {
+      instance?.off("hide");
+    };
+  }, [instance]);
+
+  // show/hide and animate the modal
+  useEffect(() => {
+    if (instance) {
+      const animate = instance?.$el.childNodes[1].animate;
+      if (show) {
+        instance?.show();
+        if (animate) instance?.$el.childNodes[1].animate?.(...getAnimationProps());
+      } else {
+        if (animate) {
+          const animation = instance?.$el.childNodes[1]?.animate(...getAnimationProps(true));
+          animation?.finished.then(() => instance?.hide());
+        } else {
+          instance?.hide();
+        }
+      }
+    }
+  }, [show, instance, getAnimationProps]);
+
+  // const hideModal = useCallback(() => {
+  //   instance.hide();
+  // }, [instance]);
+
+  const header = useMemo(() => {
+    const { id } = attr.title;
+    const header = React.Children.toArray(children).find(isModalHeader);
+    if (header) {
+      return cloneElement(header, { id, closeModal: onClose });
+    }
+
+    return (
+      <ModalHeader
+        title={title}
+        description={description}
+        closeModal={onClose}
+        id={id}
+        hideCloseButton={hideCloseButton}
+        closeButtonAriaLabel={closeButtonAriaLabel}
+      />
+    );
+  }, [children, attr, title, description, hideCloseButton, closeButtonAriaLabel]);
+
+  const content = useMemo(() => {
+    return (
+      React.Children.toArray(children).find(isModalContent) || (
+        <ModalContent>
+          {React.Children.toArray(children).filter(child => !isModalHeader(child) && !isModalFooter(child))}
+        </ModalContent>
+      )
+    );
+  }, [children]);
+
+  const footer = useMemo(() => {
+    return React.Children.toArray(children).find(isModalFooter) || null;
+  }, [children]);
+
+  const dialog = ReactDOM.createPortal(
+    <div
+      {...attr.container}
+      className={cx(styles.container, classNames.container)}
+      data-testid="monday-dialog-container"
+    >
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
+      <div
+        onClick={closeDialogIfNeeded}
+        className={cx(styles.overlay, classNames.overlay)}
+        data-testid="monday-modal-overlay"
+      />
+      <div
+        {...attr.dialog}
+        className={cx(styles.dialog, classNames.modal, {
+          [styles.default]: width === MODAL_WIDTH.DEFAULT,
+          [styles.full]: width === MODAL_WIDTH.FULL_WIDTH
+        })}
+      >
+        {header}
+        {content}
+        {footer}
+      </div>
+    </div>,
+    document.body
+  );
+
+  return <>{dialog}</>;
```

Comment:
```suggestion
  return dialog;
```

Reply:
I think we can remove this <> </> no?

Reply:
@hadasfa I thought so too, but when I remove it storybook stops showing the component props on the storybook page